controller/controller.go

```diff
@@ -141,6 +145,9 @@ func (r Request) ExtendSlices() (Request, error) {
 }
 
 func (c controller) Submit(req Request) (*task.TaskObject, error) {
+	if (len(req.Units) == 0) {
+		return nil, errgo.WithCausef(nil, maskAny(invalidArgumentError) , "invalid argument: Units must not be empty")
```

Comment:
Understood + agreed. I am happy to change it, but I find the name confusing and its missing a docstring that explains THAT. 

I have no suggestions for a name though :(